mesh.py

```diff
@@ -15,7 +15,7 @@
 '''
 class Mesh():
     def __init__(self, bounds, delta_x = 0, delta_y=0, delta_z=0,
-            default_material=None):
+            default_material=None, num_groups=0):
```

Comment:
Zero groups would be quite an unfortunate problem. We should probably make ``default_material`` the only optional attribute since the others should clearly be specified for all cases.